src/renderer/dx/DxFontInfo.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <dwrite.h>
+#include <dwrite_1.h>
+#include <dwrite_2.h>
+#include <dwrite_3.h>
+
+namespace Microsoft::Console::Render
+{
+    class DxFontInfo
+    {
+    public:
+        DxFontInfo() noexcept;
+
+        DxFontInfo(std::wstring_view familyName,
+                   DWRITE_FONT_WEIGHT weight,
+                   DWRITE_FONT_STYLE style,
+                   DWRITE_FONT_STRETCH stretch);
+
+        DxFontInfo(std::wstring_view familyName,
+                   unsigned int weight,
+                   DWRITE_FONT_STYLE style,
+                   DWRITE_FONT_STRETCH stretch);
+
+        std::wstring GetFamilyName() const;
+        void SetFamilyName(const std::wstring_view familyName);
+
+        DWRITE_FONT_WEIGHT GetWeight() const noexcept;
+        void SetWeight(const DWRITE_FONT_WEIGHT weight) noexcept;
+
+        DWRITE_FONT_STYLE GetStyle() const noexcept;
+        void SetStyle(const DWRITE_FONT_STYLE style) noexcept;
+
+        DWRITE_FONT_STRETCH GetStretch() const noexcept;
+        void SetStretch(const DWRITE_FONT_STRETCH stretch) noexcept;
+
+        void SetFromEngine(const std::wstring_view familyName,

```

Comment:
The name `SetFromEngine` is copied from `FontInfo`. I don't know why the name. This is just a "set-everything" method.

Reply:
I don't quite get it what you mean by "feeding it into a fresh constructed one" . sorry my understanding of English will go offline occasionally. 

I kinda like the way it is now because it has the same name as in `FontInfo`. One day maybe we can merge `FontInfo` with `DxFontInfo`, then this name would clarify what it is.

Reply:
It's fine. You can leave it for now. I agree that perhaps the other `FontInfo` can be unified one day.